x-pack/plugins/security_solution/server/lib/entity_analytics/entity_store/definition.ts

```diff
@@ -8,49 +8,51 @@
 import { entityDefinitionSchema, type EntityDefinition } from '@kbn/entities-schema';
 import { ENTITY_STORE_DEFAULT_SOURCE_INDICES } from './constants';
 
-export const HOST_ENTITY_DEFINITION: EntityDefinition = entityDefinitionSchema.parse({
-  id: 'ea_host_entity_store',
-  name: 'EA Host Store',
-  type: 'host',
-  indexPatterns: ENTITY_STORE_DEFAULT_SOURCE_INDICES,
-  identityFields: ['host.name'],
-  displayNameTemplate: '{{host.name}}',
-  metadata: [
-    'host.domain',
-    'host.hostname',
-    'host.id',
-    'host.ip',
-    'host.mac',
-    'host.name',
-    'host.type',
-    'host.architecture',
-  ],
-  history: {
-    timestampField: '@timestamp',
-    interval: '1m',
-  },
-  version: '1.0.0',
-});
+export const HOST_ENTITY_DEFINITION: (id: string) => EntityDefinition = (id) =>
```

Comment:
It's a bit weird to have ALL_CAPS for a function, but I thought I'd leave it because even though it depends on the `id`, it's still a hardcoded constant. Event the ID is, in a way, "hardcoded", it just has to be built depending on the entity type and space.

Reply:
I think we will be having this discussion again in one of my PRs later 😅 I am adding some more logic here and moved `getEntityDefinition` into this file.